File: Day2/solver.py

import re
# So, 12 red , 13 green and 14 blue have total of 39 

r=12
g=13
b=14

def main():
    sum=0
    with open("day_2.txt") as f:
        data=f.readlines()
        for i in data:
            a=re.match(r'Game (\d+)',i)
            if a:
                print(re.findall(r'(\d+) green',i))
                green=max(map(int,re.findall(r'(\d+) green',i)))
                red=max(map(int,re.findall(r'(\d+) red',i)))
                blue=max(map(int,re.findall(r'\b(\d+) blue',i)))
                # print("GREEN",red)
                # print("GGG",red,green,blue)
                if (green <= g and red <= r and blue <= b):
                    # print(a.group(1)) 
                    # print(a)
                    sum+=int(a.group(1))
                # print(type(green))
    print(sum) 

if __name__=='__main__' :
    main()

File: Day2/solver.js
const fs = require("fs");

let data_file=""
const r=12;
const b=14;
const g=13;
async function readfile(){
    try{
        data_file=(await fs.promises.readFile('day_2.txt',{encoding:"utf-8"})).split('\n');
    }
    catch(err){
        console.log(err);
    }
}

async function main(){
    let sum=0;
    let game_number="";
    await readfile();        
    for(let i of data_file)
    {
        dicter={}
        const regex0=new RegExp("Game\\s(\\d+):*");
        game_number=i.match(regex0)[1];
        const regex=new RegExp("\\d+ \\w+","g")
        let matchr=[...i.match(regex)]
            for (let j of  matchr){
            let [Num ,value]=j.split(' ')
            // console.log("NUM: ",Num);
            if (!dicter[value]){
            dicter[value]=-1
            }
            if (dicter[value]<+Num){
                dicter[value]=+Num;
            }
            }
            // console.log(dicter);

            if(dicter['red']<=r && dicter['blue']<=b && dicter['green']<=g){
                // console.log(game_number, "red :",dicter['red'] , dicter['blue'],dicter['green'], " green hai ");
                sum+=(+game_number);
            }   
 
    }
    console.log(sum)
    // console.log(data_file);
}
main()